src/runtime/hexagon_remote/sim_remote.cpp

```diff
@@ -120,12 +175,46 @@ int halide_do_task(void *user_context, halide_task_t f, int idx,
 
 int halide_do_par_for(void *user_context, halide_task_t f,
                       int min, int size, uint8_t *closure) {
-    for (int x = min; x < min + size; x++) {
-        int result = halide_do_task(user_context, f, x, closure);
+
+    master_thread_hvx_mode = get_hvx_mode();
+    int num_worker_threads = 1;
+    if (master_thread_hvx_mode == DOUBLE_VECTOR_MODE) {
+        num_worker_threads = 3;
+    }
+    int start_min = min;
+    int chunk_size = size / (num_worker_threads + 1);
+    int hw_thread = 1;
+    for (int i = 0; i < num_worker_threads; i++) {
+        work_queue[i].user_context = user_context;
+        work_queue[i].closure = closure;
+        work_queue[i].min = start_min;
+        work_queue[i].max = start_min + chunk_size;
+        work_queue[i].result = 0;
+        work_queue[i].f = f;
+        start_min += chunk_size;
+        thread_create(thread_server, &stack[i][STACK_SIZE], hw_thread++, (void *)i);
```

Comment:
This will reuse the same stacks for different threads in the case of nested parallelism...